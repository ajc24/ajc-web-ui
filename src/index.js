/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
