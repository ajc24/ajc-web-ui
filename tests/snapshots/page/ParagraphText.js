import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { ParagraphText } from '../../../src';

describe('ParagraphText', () => {
  describe('Default props and rendering - Default style with left aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphText>
            Test paragraph text.
          </ParagraphText>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Default style with center aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphText alignment="center">
            Test paragraph text.
          </ParagraphText>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Default style with right aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphText alignment="right">
            Test paragraph text.
          </ParagraphText>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Bold style with left aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphText style="bold" alignment="left">
            Test paragraph text.
          </ParagraphText>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Italics style with left aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphText style="italics" alignment="left">
            Test paragraph text.
          </ParagraphText>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
