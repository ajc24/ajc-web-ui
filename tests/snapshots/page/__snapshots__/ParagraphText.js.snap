// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParagraphText Default props and rendering - Default style with left aligned text verifies the snapshot for the component 1`] = `
<p
  className="ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-left"
>
  Test paragraph text.
</p>
`;

exports[`ParagraphText Transferred props and rendering - Bold style with left aligned text verifies the snapshot for the component 1`] = `
<p
  className="ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-left"
>
  <b>
    Test paragraph text.
  </b>
</p>
`;

exports[`ParagraphText Transferred props and rendering - Default style with center aligned text verifies the snapshot for the component 1`] = `
<p
  className="ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-center"
>
  Test paragraph text.
</p>
`;

exports[`ParagraphText Transferred props and rendering - Default style with right aligned text verifies the snapshot for the component 1`] = `
<p
  className="ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-right"
>
  Test paragraph text.
</p>
`;

exports[`ParagraphText Transferred props and rendering - Italics style with left aligned text verifies the snapshot for the component 1`] = `
<p
  className="ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-left"
>
  <i>
    Test paragraph text.
  </i>
</p>
`;
